KarhooUISDK/Screens/BookingScreen/NewQuoteList/NewQuoteListPresenter.swift

```diff
@@ -0,0 +1,49 @@
+//  
+//  NewQuoteListPresenter.swift
+//  KarhooUISDK
+//
+//  Created by Aleksander Wedrychowski on 23/02/2022.
+//  Copyright © 2022 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+import KarhooSDK
+
+enum QuoteListStete {
+    case loading
+    case fetched(quotes: [Quote])
+    case empty
+}
+
+class KarhooNewQuoteListPresenter: NewQuoteListPresenter {
+
+    private let router: NewQuoteListRouter
+    let onQuoteSelected: (Quote) -> Void
+    let onQuoteDetailsSelected: (Quote) -> Void
+    var onQuoteListStateUpdated: ((QuoteListStete) -> Void)?
+    var state: QuoteListStete = .empty
+
+    init(
+        router: NewQuoteListRouter,
+        quotes: [Quote],
+        onQuoteSelected: @escaping (Quote) -> Void,
+        onQuoteDetailsSelected: @escaping (Quote) -> Void
+    ) {
+        self.router = router
+        self.state = quotes.isEmpty ? .empty : .fetched(quotes: quotes)
+        self.onQuoteSelected = onQuoteSelected
+        self.onQuoteDetailsSelected = onQuoteDetailsSelected
+    }
+
+    func viewDidLoad() {
```

Comment:
I'd like to keep it 
- to make sure if at some point there is a need to add lifecycle-depending event, the proper endpoint is already there, so we do not need to face any inconsistencies in the future 
- because view should not know if a presenter performs any action at this point of lifecycle or not
- I tend to keep architectural elements in place even if they are not used at the moment. This approach is also suggested by creators of Clean Swift/Viper architectures. This way the development team is reminded the architecture is required and one should not be messing up with it. On the other hand, whenever some component/element is needed (like `setupProperties`, because suddenly we need to change main view properties, or change ViewController title in navigation stack) the designed place for it is ready for action.